packages/blob-index/src/sharded-dag-index.js

```diff
@@ -165,6 +176,14 @@ export const create = (content) => new ShardedDAGIndex(content)
  * @returns {Promise<API.Result<Uint8Array>>}
  */
 export const archive = async (model) => {
+  // Check if we're dealing with a large dataset
+  const totalEntries = model.shards.size
+
+  if (totalEntries > LARGE_DATASET_ARCHIVE_THRESHOLD) {
+    return await archiveLargeDataset(model)
```

Comment:
Why would we keep both implementations? I'd replace with this one unless there is a performance penalty.

Reply:
Good point. I guess I'll replace it with the new one. I haven't run a performance check, but I was thinking more about preserving the existing behavior for smaller sets, given that using batches adds some overhead in general.